runtime/landlock.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#define _GNU_SOURCE
+#include <errno.h>
+#include <fcntl.h>
+#include <linux/landlock.h>
+#include <linux/prctl.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/prctl.h>
+#include <sys/stat.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+#include "strv.h"
+
+#ifndef landlock_create_ruleset
```

Comment:
If this symbol is defined as a function rather than a macro, will this work? What is the point of these `ifndef`s?